### PR TITLE
{Application-insights} Migrate application-insights to AAZ

### PR DESCRIPTION
**Pull request by Jing-song**
---
AAZ model: https://github.com/Azure/aaz/pull/290

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command

`az monitor application-insights`

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Application-insights

File: src/application-insights/azext_applicationinsights/azext_metadata.json [hunk 1]
```diff
@@ -1,4 +1,4 @@
 {
-    "azext.minCliCoreVersion": "2.38.0",
+    "azext.minCliCoreVersion": "2.50.0",
     "azext.isPreview": true
```
Comment (jsntcy) [inline src/application-insights/azext_applicationinsights/azext_metadata.json:3]:
```suggestion
```

File: src/application-insights/setup.py [hunk 2]
```diff
@@ -8,7 +8,7 @@
 from codecs import open
 from setuptools import setup, find_packages
 
-VERSION = "0.1.19"
+VERSION = "1.0.0a1"
```
Comment (jsntcy) [inline src/application-insights/setup.py:11]:
```suggestion
VERSION = "1.0.0"
```

File: src/application-insights/HISTORY.rst [hunk 3]
```diff
@@ -2,6 +2,11 @@
 
 Release History
 ===============
+
+1.0.0a1
+++++++++++++++++++
```
Comment (jsntcy) [inline src/application-insights/HISTORY.rst:7]:
```suggestion
++++++++++++++++++
* Migrate manual code to automatic code by aaz tool.
```

**Review by jsntcy — APPROVED** [review]

**Pull request merged**